tests/codeception/_config/web.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @copyright Copyright &copy; Kartik Visweswaran, communityii, 2014 - 2015
+ * @package communityii/yii2-user
+ * @version 1.0.0
+ *
+ * @author derekisbusy https://github.com/derekisbusy
+ * @author kartik-v https://github.com/kartik-v
+ */
+
+use comyii\user\components\Buttons;
+use comyii\user\components\Views;
+use comyii\user\tests\codeception\_app\models\TestUser;
+
+return [
+    'modules' => [
+        'user' => [
+            'class' => 'comyii\user\Module',
+//            'installAccessCode' => '1234',
+            // setup your module preferences
+            // refer all module settings in the docs
+            'components' => [
+                'layouts' => [
+                    'class' => 'comyii\user\components\Layouts',
+                    'layouts' => [
+                        Views::VIEW_PROFILE_INDEX => '@frontend/views/layouts/patient',
+                        Views::VIEW_LOGIN => 'custom',
+                    ]
+                ],
+                'views' => [
+                    'class' => 'comyii\user\components\Views',
+                    'views' => [
+                    ]
+                ],
+                'actions' => [
+                    'class' => 'comyii\user\components\Actions',
+                    'actions' => [
+                        
+                    ]
+                ],
+                'buttons' => [
+                    'buttons' => [
+                        Buttons::BTN_RESET_FORM => null,
+                    ],
+                ]
+            ],
+            'types' => [
+                TestUser::TYPE_VENDOR => [
+                    'route' => 'vendor',
+                    'layouts' => [
+                        'layouts' => [
```

Comment:
Yeah, I didn't like the naming either. These could be simple arrays instead of components but I thought components would be better since you can attach events and behaviors and add methods if needed. Since all the views, buttons etc are numbers maybe we can override the __set() and __get() method to say `if (is_int($name))` store in the array container otherwise treat as a property.

```php
public function __set($name, $value) {
    if (is_int($name)) {
        $this->layouts[$name] = $value;
    } else {
        $this->{$name} = $value;
    }
}
```

Then the config would look like:

```php
'layouts' => [
    'class' => '...',
    Views::VIEW_PROFILE_INDEX => '@frontend/views/layouts/customer',
]
```